front/app/components/admin/PostManager/InputManager.tsx

```diff
@@ -0,0 +1,369 @@
+import React, { ChangeEvent, Suspense, lazy, useEffect, useState } from 'react';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import { DndProvider } from 'react-dnd';
+import { isNilOrError } from 'utils/helperUtils';
+import useIdeaStatuses from 'api/idea_statuses/useIdeaStatuses';
+import useTopics from 'api/topics/useTopics';
+import useProjectAllowedInputTopics from 'api/project_allowed_input_topics/useProjectAllowedInputTopics';
+import { getTopicIds } from 'api/project_allowed_input_topics/util/getProjectTopicsIds';
+import {
+  LeftColumn,
+  MiddleColumn,
+  MiddleColumnTop,
+  PreviewMode,
+  Sticky,
+  StyledExportMenu,
+  StyledInput,
+  ThreeColumns,
+  TopActionBar,
+} from '.';
+import ActionBar from './components/ActionBar';
+import FilterSidebar from './components/FilterSidebar';
+import PostTable from './components/PostTable';
+import InfoSidebar from './components/InfoSidebar';
+import IdeasCount from './components/IdeasCount';
+import FeedbackToggle from './components/TopLevelFilters/FeedbackToggle';
+const LazyPostPreview = lazy(
+  () => import('components/admin/PostManager/components/PostPreview')
+);
+import Outlet from 'components/Outlet';
+import { useSearchParams } from 'react-router-dom';
+import { removeSearchParams } from 'utils/cl-router/removeSearchParams';
+import { IProjectData } from 'api/projects/types';
+import { TPhases } from 'api/phases/types';
+import useIdeas from 'api/ideas/useIdeas';
+import { IQueryParameters, Sort } from 'api/ideas/types';
+import { getPageNumberFromUrl, getSortDirection } from 'utils/paginationUtils';
+
+interface Props {
+  // When the PostManager is used in /admin/projects, we pass down the current project id as a prop
+  projectId?: string | null;
+  visibleFilterMenus: TFilterMenu[]; // cannot be empty.
+  defaultFilterMenu: TFilterMenu;
+  phases?: TPhases;
+  // When the PostManager is used in /admin/ideas, the parent component passes
+  // down the array of projects the current user can moderate.
+  projects?: IProjectData[] | null;
+}
+
+export type TFilterMenu = 'topics' | 'phases' | 'projects' | 'statuses';
+
+const InputManager = ({
+  defaultFilterMenu,
+  visibleFilterMenus,
+  projects,
```

Comment:
Would it be better if we only have a project prop and if we pass only 1 item to it, we know it's the `ProjectIdeas` variant of the component? Having 2 props for this is confusing to me.

Reply:
I asked in #team-product whether we should kill the general ('AllIdeas') version. Awaiting product's take before doing more work on this.